### DISCUSSION TITLE
Convert heic to jpeg with same quality

### DISCUSSION THREAD

**Discussion by vikasjio** [libvips/libvips#4030] [General]
I am developing a service that take heic image and convert it to jpeg.
I can load heic and use vips_jpegsave_mime() for this. However this method uses default Q factor of 75, irrespective of the heic/source quality - This approach has dual challenges of increased size for very low quality heic and reduced jpeg quality for high quality heic. So I intend to provide specific/right/original value for this Q param.
However I did not find anything to get such Quality factor from heic VipsImage.

So i solicit help to either 1. get heic quality factor Or 2. some jpegsave method that uses quality factor from source/VipsImage instead of using default 75. or vips_jpegsave_mime() has some special value (e.g. -1) for this?

I understand both formats/codecs may have different methods for quality determination/estimation, however a rough estimate value may be helful for me.

**Comment by jcupitt** [comment]
Hello @vikasjio,

I don't think it's possible to determine the encode quality a HEIC used. This is true for almost all image formats -- they standardise the file format, not the specific encoder implementation, and that means you can't recover the encoder settings from the encoded file.

You could experimentally reencode the files with various HEIC Q factors and see which gets you a similar file size, but of course that can be VERY slow. You might need to encode four times to get Q +/- about 5.
